### PR TITLE
feat: drop creators to artists offchain profiles & base calendar drops

### LINKED ISSUE #10359: Switch drop creators to artists offchain profiles (created by)
This took us quite a long time, but we should slowly switch the drops to show the actual artist

image

will drop how it will look

### LINKED ISSUE #10498: Drop Calendar: Possibility to have Base drops in calendar
I have added two base drops to Cloudflare so we can see them in the Drops Calendar.
Screenshot 2024-06-19 at 10 59 32

In the chain field, I added the _base_, and in the external link, I added just a l_ink to our Farcaster channel_.
Screenshot 2024-06-19 at 10 59 39

Drops are still not showing up in the calendar. Not sure what needs to be adjusted. But it would be nice if these drops could be on the Calendar. Viki told me to assign you to it Hassan.

### LINKED ISSUE #10533: drop card v2.5 - artist on card
image

# file 
https://www.figma.com/design/JXNtryaNYhY0XJNArzvPQx/Handoff?node-id=394-6126&t=dLtMjF81b0yy56Fg-11

- added artist name and profile picture on card 

## Context
waiting for: 
- https://github.com/kodadot/nft-gallery/issues/10359

### PR DESCRIPTION
**Pull request by hassnian**
## PR Type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

## Context

❗️❗️ Needs https://github.com/kodadot/private-workers/pull/191

- [x] Closes #10359
- [x] Closes #10533
- [x] Closes #10498

## Needs QA check

- @kodadot/qa-guild please review

## Needs Design check

- @exezbcz please review

#### Did your issue had any of the "$" label on it?

- [x] Fill up your DOT address: [Payout](https://kodadot.xyz/dot/transfer?target=16faLfsywwNATaEfbH2ah75dn6ZmctQWpMS5G4KFhbmj5hnD)

## Screenshot 📸

- [x] My fix has changed **something** on UI; 

### profile hover card

CleanShot 2024-06-24 at 11 09 39

### CODE REVIEW

**Comment by Jarsen136** [comment]
I don't know if the PR solves the issue. From the description of the original issue, it suppose to show the real artist of the drop. We may need a new field for every drop to customise the `artist` address.

image

**Comment by hassnian** [comment]
> I don't know if the PR solves the issue. From the description of the original issue, it suppose to show the real artist of the drop. We may need a new field for every drop to customise the `artist` address.

added new field `artist_address`  https://github.com/kodadot/private-workers/pull/189

should we hide the created by if `artist_address` is not provided , or should we fallback on the collection `currentOwner` 

cc @exezbcz

**Comment by exezbcz** [comment]
> should we hide the created by if `artist_address` is not provided , or should we fallback on the collection `currentOwner`

we will make sure artist always drop their address

**Comment by exezbcz** [comment]
the list of the artists is here: https://github.com/kodadot/ops-internal/issues/1718#issuecomment-2173578189

**Review by Jarsen136 — APPROVED** [review]

**Comment by hassnian** [comment]
> > should we hide the created by if `artist_address` is not provided , or should we fallback on the collection `currentOwner`
> 
> we will make sure artist always drop their address, but yeah, some mechanism of hiding could be nice

done

**Comment by hassnian** [comment]
@Jarsen136 added #10533 pls re-review 

thanks

**Review by Jarsen136 — APPROVED** [review]

**Comment by hassnian** [comment]
@JustLuuuu worker side is ready , can you update the `creator` on all drops , thanks

**Comment by vikiival** [comment]
> worker side is ready , could you update the creator field on all drops ?

This query successfully executed.

If you can drop a list of the `collection_id, recipient` for drops

**Comment by JustLuuuu** [comment]
@vikiival @hassnian  some of the older drops were not in the calendar - where should I update the "creator"? should I add those drops to cloudfare?

**Comment by exezbcz** [comment]
> 2\. base calendar drops are now visible,  @exezbcz  should we show the `chain` inside the drop calendar modal ?

like when you create calendar event?

**Comment by vikiival** [comment]
> where should I update the "creator"

Hello @JustLuuuu,


Creator is part of the `calendar` table so you can edit that as usually ^-^ 

Screenshot 2024-06-26 at 12 18 00

**Comment by JustLuuuu** [comment]
> > where should I update the "creator"
> 
> Hello @JustLuuuu,
> 
> Creator is part of the `calendar` table so you can edit that as usually ^-^

Yeah, I found this, and I uploaded creators for drops we have in the calendar—but most of the old drops are not in the calendar.

**Comment by exezbcz** [comment]
what is missing here? 👀

**Comment by exezbcz** [comment]
does not work for me yet 
image


cards are oki


does the worker need to be updated or?

**Comment by vikiival** [comment]
cc @hassnian

**Comment by hassnian** [comment]
> cc @hassnian

`creator` field has not been updated 

cc @JustLuuuu 

https://fxart.kodadot.workers.dev/drops

CleanShot 2024-07-03 at 11 12 39@2x

**Comment by vikiival** [comment]
For others @JustLuuuu has to chime in

```json
{
  "data": {
    "collectionEntities": [
      {
        "id": "52",
        "name": "Whirls",
        "recipient": "1L2xstToYKYE3m7R2qLM8MMpGQNmioSkMpx39QcxQucgnNZ"
      },
      {
        "id": "116",
        "name": "Shapes",
        "recipient": null
      },
      {
        "id": "118",
        "name": "Packing",
        "recipient": null
      },
      {
        "id": "120",
        "name": "Aurora",
        "recipient": null
      },
      {
        "id": "123",
        "name": "Aeroloom",
        "recipient": null
      },
      {
        "id": "115",
        "name": "Tilescapes",
        "recipient": "158mjiYukLQPQwz7fReGuxofQg7ssUP4zaZXVukiyDKiqEdL"
      },
      {
        "id": "121",
        "name": "Electronik",
        "recipient": null
      },
      {
        "id": "158",
        "name": "Under the Microscope",
        "recipient": "0x313373557474396555445932433174637150596433384473625378347a37785a4a3735445667686a5053585639316e66"
      },
      {
        "id": "130",
        "name": "Enough",
        "recipient": "13DgGuPvoJgtgtARyATbAEjjQWtepnC8mBjrEgzWYRSTS5mN"
      },
      {
        "id": "134",
        "name": "memoverse",
        "recipient": "123ZjciTXEfgJmQFCqhCqwwYMe6EzbFASA5t2r8nBZXbkhen"
      },
      {
        "id": "127",
        "name": ".hdd",
        "recipient": "1VshyWpZUt8mgtWKgv9pgj8Gin99Wg5PCaDL3iCABixQbUP"
      },
      {
        "id": "112",
        "name": "Convergente",
        "recipient": null
      },
      {
        "id": "76",
        "name": "Assemblies",
        "recipient": "14zMpLStFuyTSmWY4rmGRMSaneTpe5Z425ZDYdSuABoYmejW"
      },
      {
        "id": "152",
        "name": "xest",
        "recipient": "12uQawLwd83igjibocdM75qCN2GzVW8oLPaXaaUeBG9hvz3s"
      },
      {
        "id": "106",
        "name": "Refraction",
        "recipient": "13DVkU7wKdGyY5X53KzQaQhZj1EcR6CmZpjKLkknjcnhBWd7"
      },
      {
        "id": "95",
        "name": "Echo",
        "recipient": "16dDquvnAbg3LHohdJHDaB5t5Mx6NFA2afNiCm9tEoXcR8YM"
      },
      {
        "id": "77",
        "name": ".cpu",
        "recipient": "1VshyWpZUt8mgtWKgv9pgj8Gin99Wg5PCaDL3iCABixQbUP"
      },
      {
        "id": "99",
        "name": "LEADLIGHT",
        "recipient": "15kNtZhj6Wrknr8Qx2MVZacYLyorFrXLHy72FKBjuqfi4i67"
      },
      {
        "id": "91",
        "name": "SailWings",
        "recipient": "15XwKYp5BDpW18xKqPynnuggZhxsWG3sWzKQYave8UErz2rh"
      },
      {
        "id": "51",
        "name": "Chroma Lattice",
        "recipient": "139hEH4Mq9NM2AcDrY9PBZc4KRcBVsaRhAnnBNLZd5gffLgf"
      },
      {
        "id": "63",
        "name": "Genera",
        "recipient": null
      },
      {
        "id": "46",
        "name": "Snowflakes",
        "recipient": null
      },
      {
        "id": "108",
        "name": "Motifs",
        "recipient": null
      },
      {
        "id": "50",
        "name": ".motherboard",
        "recipient": null
      },
      {
        "id": "49",
        "name": "Wall Street Art",
        "recipient": "15Q7VywFCB9o6TBGBY3YJ8xWTVR22EVdRMh153FQMgAattbG"
      },
      {
        "id": "87",
        "name": "Florist",
        "recipient": null
      },
      {
        "id": "66",
        "name": "Split",
        "recipient": "1nmjxYxgXyay2YABPhUrP7HhvyKgBUqSd2R2LKE77eRAMWV"
      },
      {
        "id": "82",
        "name": "Alchemy",
        "recipient": null
      },
      {
        "id": "38",
        "name": "Pare1d0scope",
        "recipient": "1RWmphfVQAwfGAizAqy6ymaTvoc8rknHji6As1YxUWSGpta"
      },
      {
        "id": "60",
        "name": "Christmas City",
        "recipient": null
      },
      {
        "id": "40",
        "name": "Swirls",
        "recipient": null
      }
    ]
  }
}
```

**Comment by JustLuuuu** [comment]
@vikiival - added to profiles that were available

**Comment by hassnian** [comment]
@exezbcz ready to test 

CleanShot 2024-07-05 at 09 57 09@2x

CleanShot 2024-07-05 at 09 58 23@2x

**Comment by exezbcz** [comment]
image


dark mode








please could you switch the icons to white in dark mode

**Comment by vikiival** [comment]
cc @hassnian

**Comment by hassnian** [comment]
added dark mode

File: composables/useIcon.ts [hunk 1]
```diff
@@ -20,5 +22,23 @@ export const useIcon = () => {
     isDarkMode.value ? '/unlockable-dark.svg' : '/unlockable.svg',
   )
 
-  return { getTokenIconBySymbol, signUpVoucherIcon, unlockableIcon }
+  const getChainIcon = (prefix: Prefix | null): string | null => {
+    switch (prefix) {
+      case 'ahp':
+        return isDarkMode.value ? '/chain/ahp_dark.svg' : '/chain/ahp.svg'
+      case 'ahk':
+        return '/token/kusama_asset_hub.svg'
```
Comment (vikiival) [inline composables/useIcon.ts:30]: ksm does not have dark mode ?

**Review by vikiival — APPROVED** [review]

File: composables/useIcon.ts (hunk 1 above)
Comment (hassnian) [reply composables/useIcon.ts]: done 299768d

**Comment by vikiival** [comment]
@hassnian pls resolve issues

**Comment by vikiival** [comment]
cc @hassnian

**Comment by hassnian** [comment]
> cc @hassnian

done

**Pull request merged**